gst/renderer/gstttmlrender.c

```diff
@@ -467,6 +485,12 @@ gst_ttml_render_setcaps (GstTtmlRender * render, GstCaps * caps)
     GST_DEBUG_OBJECT (render, "unsupported caps %" GST_PTR_FORMAT, caps);
     ret = FALSE;
   }
+  if (ret && render->attach_compo_to_buffer) {
+    if (render->overlay_composition_width > 0 && render->overlay_composition_height > 0) {
+      render->width = render->overlay_composition_width;
+      render->height = render->overlay_composition_height;
```

Comment:
Here we essentially override the internal resolution, and composition renderer would just use the resolution detached from actual video from now on.